test/planning/test_graph_search.py

```diff
@@ -0,0 +1,66 @@
+from pyroboplan.core.utils import configuration_distance
+from pyroboplan.planning.graph import Node, Graph
+from pyroboplan.planning.graph_search import astar, dfs
+
+
+def construct_square_graph(n):
+    """
+    Creates a connected n x n square graph for testing.
+
+    Each node in the graph will be connected to its vertical and horizontal neighbor.
+    For example, for n == 3:
+
+    Graph
+    -----
+    (1,1) -- (1,2) -- (1,3)
+     |        |        |
+    (2,1) -- (2,2) -- (2,3)
+     |        |        |
+    (3,1) -- (3,2) -- (3,3)
```

Comment:
nice! I also now realize that tests don't actually render in the docs, so all good